src/server/frontend_wayland/wl_seat.cpp

```diff
@@ -228,8 +233,23 @@ void mf::WlSeat::set_focus_to(WlSurface* new_surface)
                 listener->focus_on(nullptr);
             });
     }
+    if (focused_surface)
+    {
+        focused_surface.value().remove_destroy_listener(focused_surface_destroy_listener_id);
+    }
     focused_client = new_client;
     focused_surface = mw::make_weak(new_surface);
+    if (new_surface)
+    {
+        focused_surface_destroy_listener_id = new_surface->add_destroy_listener([this]()
```

Comment:
Is there any guarantee that the `WlSeat` will outlive `new_surface`? ie: what is stopping the `this` pointer from dangling? I guess that would only happen during teardown, but it looks to me like we destroy the `WlSeat` global before we destroy the `wl_display`. I'm not sure if tearing down the `wl_display` will actually run the `wl_surface` destroy listeners, though?

Reply:
The listener is removed in the `WlSeat` destructor, and I added a comment to point that out